### DISCUSSION TITLE
Playing videos still prevents the system from entering sleep mode or activating the screensaver

### DISCUSSION THREAD

**Discussion by Taiizor** [cefsharp/CefSharp#4996] [Q&A]
Hello,

I am using CefSharp in my WPF project to play a looping video. However, as long as the window remains open, neither the screensaver activates nor does the system go to sleep.

Below is a code snippet I tried as a solution, but it didn't resolve the issue:

```csharp
[DllImport("user32.dll", SetLastError = true)]
private static extern void SetThreadExecutionState(uint esFlags);

private const uint ES_CONTINUOUS = 0x80000000;
private const uint ES_SYSTEM_REQUIRED = 0x00000001;
private const uint ES_DISPLAY_REQUIRED = 0x00000002;

private void AllowScreenSaver()
{
    SetThreadExecutionState(ES_CONTINUOUS);
}

private void PreventScreenSaver()
{
    SetThreadExecutionState(ES_CONTINUOUS | ES_DISPLAY_REQUIRED | ES_SYSTEM_REQUIRED);
}
```

Additionally, I tried using launcher arguments like `--disable-background-timer-throttling --disable-renderer-backgrounding --disable-backgrounding-occluded-windows`, but these didn’t work either.

Interestingly, the issue is resolved in MpvPlayer by setting the `stop-screensaver` option to `no`. When reviewing its code, I noticed that it uses `ES_CONTINUOUS`.

https://github.com/mpv-player/mpv/blob/a283f66ede58e0182ac8cd4c930238144427fa74/video/out/w32_common.c#L2326
```c
case VOCTRL_KILL_SCREENSAVER:
	w32->disable_screensaver = true;
	SetThreadExecutionState(ES_CONTINUOUS | ES_SYSTEM_REQUIRED | ES_DISPLAY_REQUIRED);
	return VO_TRUE;
case VOCTRL_RESTORE_SCREENSAVER:
	w32->disable_screensaver = false;
	SetThreadExecutionState(ES_CONTINUOUS);
	return VO_TRUE;
```

**Comment by amaitland** [comment]
I've never tried this, so don't have an answer for you. Only suggestion I've got is to try calling `SetThreadExecutionState` on the `CEF UI Thread`, which by default isn't the `WPF` main thread.

```c#
Cef.UIThreadTaskFactory.StartNew(delegate
{
	//DO SOMETHING
});
```

If that doesn't work then you'll probably need to ask on https://www.magpcss.org/ceforum/viewforum.php?f=6

Also uou can search the `Chromium` source online at https://source.chromium.org/chromium

**Reply by Taiizor** [reply]
> I've never tried this, so don't have an answer for you. Only suggestion I've got is to try calling `SetThreadExecutionState` on the `CEF UI Thread`, which by default isn't the `WPF` main thread.
> 
> ```cs
> Cef.UIThreadTaskFactory.StartNew(delegate
> {
> 	//DO SOMETHING
> });
> ```
> 
> If that doesn't work then you'll probably need to ask on https://www.magpcss.org/ceforum/viewforum.php?f=6
> 
> Also uou can search the `Chromium` source online at https://source.chromium.org/chromium

I tried this, but it didn't work. Most likely, video playback itself is automatically preventing sleep/screensaver in the background.

I’ll ask in the forum shortly; hopefully, there’s a solution.
